payment/src/main/java/org/killbill/billing/payment/core/sm/PaymentAutomatonRunner.java

```diff
@@ -128,8 +129,10 @@ public PaymentStateContext buildPaymentStateContext(final boolean isApiPayment,
                                                         final Iterable<PluginProperty> properties,
                                                         final CallContext callContext,
                                                         final InternalCallContext internalCallContext) throws PaymentApiException {
+
         // Retrieve the payment id from the payment external key if needed
         final UUID effectivePaymentId = paymentId != null ? paymentId : retrievePaymentId(paymentExternalKey, paymentTransactionExternalKey, internalCallContext);
+        verifyPaymentMethodExistsForRefund(effectivePaymentId, transactionType, internalCallContext);
```

Comment:
@matias-aguero-hs  At a high level, I think the use case where we try to gather the `pluginName` based on a `paymentMethodId` (whether that payment method was deleted or not) makes sense. So the fact that `getPaymentProviderPluginName` uses the `getPaymentMethodIncludedDeleted` seems correct to me.

Now regarding use cases, i can see two different use cases:
1. One use case is the 'GET' path (e.g retrieve detail about a previous payment, including payment plugin properties). Whether payment method associated with that payment was deleted or not, those properties should be returned.
2. Other use case is the creation of new payment transaction. (`REFUND` case being one of them). In all those cases, we should not allow creating new transaction with a deleted payment method.

So probably my thinking is that depending on context (maybe a new boolean) we rely or not on the deleted payment method. Does that make sense?